tools/manifest/manifest.py

```diff
@@ -279,6 +281,10 @@ def update(self, tree):
                     if old_hash != file_hash:
                         new_type, manifest_items = source_file.manifest_items()
                         hash_changed = True
+                        try:
```

Comment:
Can we not put this in a `if new_type != old_type` block? Why do we need to handle the exception here?

Reply:
Is it necessary? We unconditionally set `self._data[new_type][rel_path]` later. I'd prefer having fewer conditional switches.

Reply:
I had the same comment. I think that although it increases the cyclomatic complexity of the function it also serves as documentation as to why we're doing it.

Reply:
Done.